src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -1289,4 +1290,1373 @@ void EIRPlantLoopHeatPump::oneTimeInit(EnergyPlusData &state)
+                } else {
+                    Real64 defaultVal = 1.0;
+                    if (!state.dataInputProcessing->inputProcessor->getDefaultValue(
+                            state, cCurrentModuleObject, "maximum_part_load_ratio", defaultVal)) {
+                        // this error condition would mean that someone broke the input dictionary, not their
+                        // input file.  I can't really unit test it so I'll leave it here as a severe error
+                        // but excluding it from coverage
+                        ShowSevereError(state,                                                                   // LCOV_EXCL_LINE
+                                        "EIR PLFFHP: maximum PLR not entered and could not get default value."); // LCOV_EXCL_LINE
+                        errorsFound = true;                                                                      // LCOV_EXCL_LINE
+                    } else {
+                        thisPLHP.maxPLR = defaultVal;
+                    }
+                }
+
+                // A14 fuel_energy_input_ratio_defrost_adjustment_curve_name
+                if (thisPLHP.EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredCooling) {
+                    thisPLHP.defrostEIRCurveIndex = 0;
+                } else {
+
+                    if (fields.find("fuel_energy_input_ratio_defrost_adjustment_curve_name") != fields.end()) {
+                        auto &eirDefrostName = fields.at("fuel_energy_input_ratio_defrost_adjustment_curve_name");
+                        thisPLHP.defrostEIRCurveIndex =
+                            Curve::GetCurveIndex(state, UtilityRoutines::MakeUPPERCase(eirDefrostName.get<std::string>()));
+                        if (thisPLHP.defrostEIRCurveIndex == 0) {
+                            ShowSevereError(state,
+                                            fmt::format("Invalid curve name for EIR FFHP (name={}; entered curve name: {}",
+                                                        thisPLHP.name,
+                                                        eirDefrostName.get<std::string>()));
+                            errorsFound = true;
+                        }
+                    } else {
+                        thisPLHP.defrostEIRCurveIndex = 0;
+                    }
+                }
+                // A15 defrost_control_type
+                if (thisPLHP.EIRHPType == DataPlant::PlantEquipmentType::HeatPumpFuelFiredCooling) {
+                    thisPLHP.defrostType = DefrostType::Invalid;
+                } else {
+                    std::string defrostControlType = UtilityRoutines::MakeUPPERCase(fields.at("defrost_control_type").get<std::string>());
+                    thisPLHP.defrostType = static_cast<DefrostType>(getEnumerationValue(DefrostTypeNamesUC, defrostControlType));
+                    if (thisPLHP.defrostType == DefrostType::Invalid) {
```

Comment:
Same comments as above about `find/at` and required enumeration fields.

Reply:
Addressed in earlier changes.